### PR TITLE
bauhaus controls: pass keypress event to view if not handled

### PR DESCRIPTION
**Pull request by elstoc**
if bauhaus controls do not handle a key press event and it is not
handled by an accelerator, finally pass the keypress to the view to
handle

ensures that even while a bauhaus control has focus the darkroom
preview shortcut still functions

also changes the preview shortcut to grab focus to the centre view so
it works when pressed twice in succession

### CODE REVIEW

**Comment by elstoc** [comment]
This, along with my earlier commits, should mean that the preview and sticky preview shortcuts now work fairly consistently in the darkroom and lighttable views.

**Comment by TurboGit** [comment]
Can you summarize what it fixes ? I can't find a scenario where a key was not handled and where it is now. Maybe I'm missing something.

**Comment by elstoc** [comment]
If you change a bauhaus slider (click on it or scroll over it), the slider grabs keyboard focus. After that point, normal keyboard accelerators work fine but the preview accelerator does not, because nothing runs the key_pressed function in the darkroom view. Similarly for comboboxes.

**Comment by TurboGit** [comment]
You mean for example in lt select and grad the quality slider in export module and then do a Alt+W to enable full preview ?

**Comment by TurboGit** [comment]
I'm asking because this case does not work with or without the patch.

**Comment by elstoc** [comment]
I think in the lighttable you need to be hovering the image for it to work anyway, so this fix doesn't make a difference. Try it in the darkroom.

**Comment by TurboGit** [comment]
In darkroom with or without your patch I can click on Alt-1 to have 100% zoom after dragging sliders. So again I don't see a difference :) I have tried many things and cannot find a case where something not possible is possible now. A precise scenario would help I suppose.

**Comment by elstoc** [comment]
Zoom close-up (Alt-1) is a regular keyboard shortcut and isn't built into the key_pressed function of darkroom.c so I would expect that to work with or without the patch. 

If you drag a slider (on, say, exposure) and then hit the shortcut for the full screen preview (W), it will only work with the patch. It may be that the preview is the only thing that's affected - the other keys (left,right,up,down) will be handled by the bauhaus handler.

**Comment by TurboGit** [comment]
The `W` (full preview in darkroom) does not work for me when grabbing the exposure slider for example. I did test this explicitly, no luck. If I click on the center image and press `W` then the display is fine. Back to any slider and pressing `W` does not work. Now what could be the cause of the discrepancies between us ???

**Comment by TurboGit** [comment]
Maybe a DE issue ? I'm using GNOME Shell 3.36.2 on GNU/Debian.

**Comment by elstoc** [comment]
XFCE on ArchLinux. Bleeding edge distro so it'll have the latest version of everything. What you describe (clicking the centre image) is what I always used to have to do without this patch.

**Comment by elstoc** [comment]
I haven't #included ``control.h`` in ``bauhaus.c``. Could that be it (though why it works for me I don't know)?

**Comment by Mark-64** [comment]
Form me in Ubuntu 20.04 the `W` only works once in darkroom. Then the center image lose focus and I have to click on the image to make it work again.

**Comment by elstoc** [comment]
@Mark-64 that's the second thing fixed by this patch.

**Comment by elstoc** [comment]
I'm working in the dark a little here but I've tried something else if anyone can test.

**Comment by elstoc** [comment]
@johnny-bit reports that this works now

**Comment by johnny-bit** [comment]
after recent change the issue is fixed, i can press "W" whilst keeping mosuse button pressed/dragging slider even!

**Comment by TurboGit** [comment]
Works as expected now ! Thanks.
